### PR TITLE
Fix situation when iot workspace file has different config file name

### PR DESCRIPTION
**Pull request by dilin-MS**

### CODE REVIEW

File: src/Models/IoTWorkspaceProject.ts [hunk 1]
```diff
@@ -185,14 +185,38 @@ export class IoTWorkspaceProject extends IoTWorkbenchProjectBase {
...
 
-  async openProject(
-      scaffoldType: ScaffoldType, openInNewWindow: boolean,
-      openScenario: OpenScenario): Promise<void> {
+  /**
+   * Validate workspace config file exists.
+   * Handles situation when workspace config file has a different name.
+   * @param scaffoldType
+   */
+  async validateWorkspaceConfigFilePath(scaffoldType: ScaffoldType):
```
Comment (eriolchan) [inline src/Models/IoTWorkspaceProject.ts]: It seems this function is try to assign workspace file to variable this.workspaceConfigFilePath? The logic is to find files with specified suffix in folder. It throws error if there is no file found, what if there are multiple file found?
BTW, you can use this method (vscode.workspace.findFiles) to find files.

Comment (dilin-MS) [reply src/Models/IoTWorkspaceProject.ts]: If there is many, we use the first file found.

File: src/Models/IoTWorkspaceProject.ts [hunk 2]
```diff
@@ -185,14 +185,38 @@ export class IoTWorkspaceProject extends IoTWorkbenchProjectBase {
...
+      Promise<void> {
     if (!await FileUtility.fileExists(
             scaffoldType, this.workspaceConfigFilePath)) {
-      throw new Error(`Workspace config file ${
-          this.workspaceConfigFilePath} does not exist. Please initialize the project first.`);
+      // Handles situation when workspace config file has a different name.
```
Comment (dooriya) [inline src/Models/IoTWorkspaceProject.ts]: Do not assume that the workspace file name is the same as the root folder name.

File: src/Models/IoTWorkspaceProject.ts [hunk 3]
```diff
@@ -185,14 +185,38 @@ export class IoTWorkspaceProject extends IoTWorkbenchProjectBase {
...
-      openScenario: OpenScenario): Promise<void> {
+  /**
+   * Validate workspace config file exists.
+   * Handles situation when workspace config file has a different name.
+   * @param scaffoldType
+   */
+  async validateWorkspaceConfigFilePath(scaffoldType: ScaffoldType):
+      Promise<void> {
     if (!await FileUtility.fileExists(
             scaffoldType, this.workspaceConfigFilePath)) {
```
Comment (dooriya) [inline src/Models/IoTWorkspaceProject.ts]: We may need to revisit the APIs in FileUtility, I don't think a general file utility API should care about the scaffold type.

**Pull request closed without merging**